src/components/tab-bar.component.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-unused-expressions */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Animated, LayoutAnimation } from 'react-native';
+import { TabBarBottom, NavigationActions } from 'react-navigation';
+
+import { animations } from 'config';
+
+const mapStateToProps = state => {
+  return {
+    isTabBarVisible: state.auth.isTabBarVisible,
+  };
+};
+
+class TabBarComponent extends Component {
+  props: {
+    isTabBarVisible: boolean,
+    navigation: Object,
+    jumpToIndex: Function,
+  };
+
+  state: {
+    animatedHeight: Animated,
+    expandedHeight: number,
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      animatedHeight: new Animated.Value(),
+      expandedHeight: '',
+    };
+  }
+
+  componentWillReceiveProps(nextState) {
+    if (nextState.isTabBarVisible !== this.props.isTabBarVisible) {
+      nextState.isTabBarVisible ? this.show() : this.hide();
+    }
+  }
+
+  setExpandedHeight(event) {
+    if (!this.state.expandedHeight) {
+      const expandedHeight = event.nativeEvent.layout.height;
+
+      this.state = {
+        animatedHeight: new Animated.Value(expandedHeight),
+        expandedHeight,
+      };
+    }
+  }
+
+  changeHeight(from, to) {
```

Comment:
You should change this to `changeHeight = (from, to) => { ... }` and avoid calling arrow functions in the render methods.